projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,169 @@
+  @Input() displayValue?: ReadLinkValue;
+  @Input() parentResource: ReadResource;
+  @Input() propType: string;
+  resources: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Displays a selected resource using its label.
+   *
+   * @param resource the resource to be displayed (or no selection yet).
+   * @returns
+   */
+  displayResource(resource: ReadResource | null): string {
+
+    // null is the initial value (no selection yet)
+    if (resource !== null) {
+      return resource.label;
+    }
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+  searchByLabel(searchTerm: string): ReadResource[] {
+    const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propType);
+    if (typeof linkType === 'string') {
+      this.restrictToResourceClass = this.parentResource.entityInfo.properties[linkType].objectType;
+    }
+    // at least 3 characters are required
+    if (searchTerm.length >= 3) {
+
+      this.knoraApiConnection.v2.search.doSearchByLabel(searchTerm, 0, {limitToResourceClass: this.restrictToResourceClass} ).subscribe(
+        (response: ReadResource[]) => {
+          this.resources = response;
+        });
+    } else {
+      // clear selection
+      this.resources = undefined;
+
+    }
+    return this.resources;
+  }
+
+  // show the label of the linked resource
+  getInitValue(): ReadResource | null {
+    if (this.displayValue !== undefined) {
+      return this.displayValue.linkedResource;
+    } else {
+      return null;
+    }
+  }
+
+  standardValidatorFunc: (val: any, comment: string, commentCtrl: FormControl) => ValidatorFn = (initValue: any, initComment: string, commentFormControl: FormControl): ValidatorFn => {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+
+      const invalid = (!(control.value instanceof ReadResource) || initValue.id === control.value.id)
+        && (initComment === commentFormControl.value || (initComment === null && commentFormControl.value === ''));
+
+      return invalid ? {valueNotChanged: {value: control.value}} : null;
+    };
+  };
+
+  // override the resetFormControl() from the base component to deal with initial link value label
+  resetFormControl(): void {
+    super.resetFormControl();
+
+    if (this.valueFormControl !== undefined) {
+      if (this.mode !== 'read') {
+        this.valueFormControl.setValue('');
```

Comment:
I think this should be removed because the user probably just wants to change the comment.

Reply:
if the user only changes comment, the initial value is set back. See 
` // add IRI to updatedLinkValue only if the user submitted a new Link value
    if (this.valueFormControl.value !== null && this.valueFormControl.value !== '') {
      updatedLinkValue.linkedResourceIri = this.valueFormControl.value.id;
    } else {
      // if user just updated comment value keep the old link value
      updatedLinkValue.linkedResourceIri = this.displayValue.linkedResourceIri;
    }`

Reply:
Since the existing value is initialised, couldn't you just leave it until the user decides to change it (or not)?

Reply:
see also https://github.com/dasch-swiss/knora-ui-ng-lib/pull/27#pullrequestreview-376625522